app/Http/Controllers/Users/WorklogController.php

```diff
@@ -46,21 +62,31 @@ public function create()
      */
     public function store(StoreWorklogRequest $request)
     {
-        Worklog::create($request->validated());
-
+        try{
+            $this->worklogService->createWorklog($request->validated());
+        }catch(WorklogNotCreatedException $exception){
+            return back()
+                ->with('error', $exception->getMessage());
+        }
         return redirect()
             ->route('worklogs.index')
-            ->with('success', 'Worklog created successfully.');
+            ->with('success', FlashMessages::SUCCESS_CREATE_WORKLOG);
     }
 
     /**
      * Show the form for editing the specified resource.
      *
-     * @param Worklog $worklog
-     * @return Factory|View|Response
+     * @param $id
+     * @return Factory|View|RedirectResponse|Response
      */
-    public function edit(Worklog $worklog)
+    public function edit($id)
     {
+        try{
+            $worklog = $this->worklogService->getWorklog($id);
```

Comment:
$this->worklogService->getWorklog($id); -> this one is already throwing not found exception

Reply:
here, the model not found exception thrown redirects to Laravel custom 404 page, but to show that worklog with the ID does not exist I have used the try-catch block to show the error message in the same view rather than Laravel custom 404 page.